File: code/JieLi_Home_Demo/NewJieliZhiNeng/GlobalViews/AppSettingVC/AboutVC/DebugView/ShareLog/TipsDeleteView.m
//
//  TipsDeleteView.m
//  JL_OTA
//
//  Created by EzioChan on 2022/10/17.
//  Copyright © 2022 Zhuhia Jieli Technology. All rights reserved.
//

#import "TipsDeleteView.h"

@interface TipsDeleteView()

@property(nonatomic,strong)UIImageView *bgView;
@property(nonatomic,strong)UIView      *centerView;
@property(nonatomic,strong)UILabel     *mainLab;
@property(nonatomic,strong)UIButton    *cancelBtn;
@property(nonatomic,strong)UIButton    *confirmBtn;
@property(nonatomic,strong)UIView      *line0;
@property(nonatomic,strong)UIView      *line1;


@end

@implementation TipsDeleteView

-(instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if(self){
        [self initUI];
    }
    return self;
}


-(void)initUI{
    self.backgroundColor = [UIColor clearColor];
    _bgView = [UIImageView new];
    _bgView.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.3];
    [self addSubview:_bgView];
    
    [_bgView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self.mas_top);
        make.bottom.equalTo(self.mas_bottom);
        make.left.equalTo(self.mas_left);
        make.right.equalTo(self.mas_right);
    }];
    
    _centerView = [UIView new];
    _centerView.backgroundColor = [UIColor whiteColor];
    _centerView.layer.cornerRadius = 15;
    _centerView.layer.masksToBounds = YES;
    [self addSubview:_centerView];
    
    [_centerView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.height.offset(130);
        make.left.equalTo(self.mas_left).offset(45);
        make.right.equalTo(self.mas_right).offset(-45);
        make.centerY.offset(-70);
    }];
    
    
    _mainLab = [UILabel new];
    _mainLab.font = FontMedium(16);
    _mainLab.textColor = [UIColor colorFromHexString:@"#242424"];
    _mainLab.text = [NSString stringWithFormat:@"%@",@"Weather delete log files or not?"];
    _mainLab.textAlignment = NSTextAlignmentCenter;
    [_centerView addSubview:_mainLab];
    [_mainLab mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(_centerView.mas_top).offset(32);
        make.height.offset(32);
        make.centerX.offset(0);
    }];
    
    
    _cancelBtn = [UIButton new];
    [_cancelBtn setTitle:@"Cancel" forState:UIControlStateNormal];
    [_cancelBtn setTitleColor:[UIColor colorFromHexString:@"#558CFF"] forState:UIControlStateNormal];
    [_cancelBtn setTitleColor:[UIColor colorFromHexString:@"#A4A4A4"] forState:UIControlStateHighlighted];
    [_cancelBtn addTarget:self action:@selector(cancelBtnAction) forControlEvents:UIControlEventTouchUpInside];
    [_centerView addSubview:_cancelBtn];
    
    _confirmBtn = [UIButton new];
    [_confirmBtn setTitle:@"Delete" forState:UIControlStateNormal];
    [_confirmBtn setTitleColor:[UIColor colorFromHexString:@"#558CFF"] forState:UIControlStateNormal];
    [_confirmBtn setTitleColor:[UIColor colorFromHexString:@"#A4A4A4"] forState:UIControlStateHighlighted];
    [_confirmBtn addTarget:self action:@selector(confirmBtnAction) forControlEvents:UIControlEventTouchUpInside];
    [_centerView addSubview:_confirmBtn];
    
    [_cancelBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(_centerView.mas_bottom).offset(0);
        make.left.equalTo(_centerView.mas_left).offset(0);
        make.right.equalTo(_confirmBtn.mas_left).offset(0);
        make.height.offset(50);
        make.width.equalTo(_confirmBtn.mas_width);
    }];
    
    [_confirmBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(_centerView.mas_bottom).offset(0);
        make.left.equalTo(_cancelBtn.mas_right).offset(0);
        make.right.equalTo(_centerView.mas_right).offset(0);
        make.height.offset(50);
        make.width.equalTo(_cancelBtn.mas_width);
    }];
    
    _line0 = [UIView new];
    _line0.backgroundColor = [UIColor colorFromHexString:@"#F5F5F5"];
    [_centerView addSubview:_line0];
    _line1 = [UIView new];
    _line1.backgroundColor = [UIColor colorFromHexString:@"#F5F5F5"];
    [_centerView addSubview:_line1];
    
    [_line0 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(_cancelBtn.mas_top).offset(0);
        make.left.equalTo(_centerView.mas_left).offset(0);
        make.right.equalTo(_centerView.mas_right).offset(0);
        make.height.offset(1);
    }];
    
    [_line1 mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(_centerView.mas_bottom).offset(0);
        make.left.equalTo(_cancelBtn.mas_right).offset(0);
        make.height.offset(50);
        make.width.offset(1);
    }];
    
}

-(void)setHidenStatus:(BOOL)hidenStatus{
    _hidenStatus = hidenStatus;
    self.hidden = hidenStatus;
}

-(void)cancelBtnAction{
    self.hidenStatus = YES;
}
-(void)confirmBtnAction{
    NSString *basicPath = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, true) lastObject];
    NSFileManager *fileManager = [NSFileManager defaultManager];
    NSArray *files = [fileManager contentsOfDirectoryAtPath:basicPath error:nil];
    for (NSString *path in files) {
        if([path hasSuffix:@".txt"]){
            NSString *newPath = [NSString stringWithFormat:@"%@/%@",basicPath,path];
            [fileManager removeItemAtPath:newPath error:nil];
        }
    }
    self.hidenStatus = YES;
}

@end

File: code/JieLi_Home_Demo/NewJieliZhiNeng/GlobalViews/AppSettingVC/AboutVC/DebugView/ShareLog/TipsDeleteView.swift
import UIKit
import SnapKit
import Combine

class TipsDeleteView: UIView {
    
    @Published var isHidenStatus: Bool = true {
        didSet {
            self.isHidden = isHidenStatus
        }
    }

    
    private let bgView = UIImageView()
    private let centerView = UIView()
    private let mainLab = UILabel()
    private let cancelBtn = UIButton()
    private let confirmBtn = UIButton()
    private let line0 = UIView()
    private let line1 = UIView()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        initUI()
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    private func initUI() {
        self.backgroundColor = .clear
        
        bgView.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.3)
        addSubview(bgView)
        bgView.snp.makeConstraints { make in
            make.edges.equalToSuperview()
        }
        
        centerView.backgroundColor = .white
        centerView.layer.cornerRadius = 15
        centerView.layer.masksToBounds = true
        addSubview(centerView)
        centerView.snp.makeConstraints { make in
            make.height.equalTo(130)
            make.left.right.equalToSuperview().inset(45)
            make.centerY.equalToSuperview().offset(-70)
        }
        
        mainLab.font = UIFont.systemFont(ofSize: 16, weight: .medium)
        mainLab.textColor = UIColor(fromHexString: "#242424")
        mainLab.text = "Weather delete log files or not?"
        mainLab.textAlignment = .center
        centerView.addSubview(mainLab)
        mainLab.snp.makeConstraints { make in
            make.top.equalTo(centerView.snp.top).offset(32)
            make.height.equalTo(32)
            make.centerX.equalToSuperview()
        }
        
        cancelBtn.setTitle("Cancel", for: .normal)
        cancelBtn.setTitleColor(UIColor(fromHexString: "#558CFF"), for: .normal)
        cancelBtn.setTitleColor(UIColor(fromHexString: "#A4A4A4"), for: .highlighted)
        cancelBtn.addTarget(self, action: #selector(cancelBtnAction), for: .touchUpInside)
        centerView.addSubview(cancelBtn)
        
        confirmBtn.setTitle("Delete", for: .normal)
        confirmBtn.setTitleColor(UIColor(fromHexString: "#558CFF"), for: .normal)
        confirmBtn.setTitleColor(UIColor(fromHexString: "#A4A4A4"), for: .highlighted)
        confirmBtn.addTarget(self, action: #selector(confirmBtnAction), for: .touchUpInside)
        centerView.addSubview(confirmBtn)
        
        cancelBtn.snp.makeConstraints { make in
            make.bottom.left.equalToSuperview()
            make.right.equalTo(confirmBtn.snp.left)
            make.height.equalTo(50)
            make.width.equalTo(confirmBtn.snp.width)
        }
        
        confirmBtn.snp.makeConstraints { make in
            make.bottom.right.equalToSuperview()
            make.left.equalTo(cancelBtn.snp.right)
            make.height.equalTo(50)
            make.width.equalTo(cancelBtn.snp.width)
        }
        
        line0.backgroundColor = UIColor(fromHexString: "#F5F5F5")
        centerView.addSubview(line0)
        line1.backgroundColor = UIColor(fromHexString: "#F5F5F5")
        centerView.addSubview(line1)
        
        line0.snp.makeConstraints { make in
            make.bottom.equalTo(cancelBtn.snp.top)
            make.left.right.equalToSuperview()
            make.height.equalTo(1)
        }
        
        line1.snp.makeConstraints { make in
            make.bottom.equalToSuperview()
            make.left.equalTo(cancelBtn.snp.right)
            make.height.equalTo(50)
            make.width.equalTo(1)
        }
    }
    
    @objc private func cancelBtnAction() {
        isHidenStatus = true
    }
    
    @objc private func confirmBtnAction() {
        if let basicPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).last {
            let fileManager = FileManager.default
            if let files = try? fileManager.contentsOfDirectory(atPath: basicPath) {
                for path in files where path.hasSuffix(".txt") {
                    let newPath = (basicPath as NSString).appendingPathComponent(path)
                    try? fileManager.removeItem(atPath: newPath)
                }
            }
        }
        isHidenStatus = true
    }
}
